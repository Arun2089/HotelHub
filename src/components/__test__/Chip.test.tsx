import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import ChipComponent from "../Chip";

const onClickMock = jest.fn();

describe("ChipComponent", () => {
  it("snapshot test", () => {
    const { asFragment } = render(
      <ChipComponent text="Test Chip" type="default" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders chip with text", () => {
    render(<ChipComponent text="Test Chip" type="default" />);
    const chipElement = screen.getByText("Test Chip");
    expect(chipElement).toBeInTheDocument();
  });

  it("calls onClick when chip is clicked", () => {
    render(
      <ChipComponent text="Test Chip" type="default" onClick={onClickMock} />
    );
    const chipElement = screen.getByText("Test Chip");
    fireEvent.click(chipElement);
    expect(onClickMock).toHaveBeenCalled();
  });

  it("renders chip with icon", () => {
    render(
      <ChipComponent
        text="Test Chip"
        type="default"
        icon={<span data-testid="icon" />}
      />
    );
    const iconElement = screen.getByTestId("icon");
    expect(iconElement).toBeInTheDocument();
  });
});
