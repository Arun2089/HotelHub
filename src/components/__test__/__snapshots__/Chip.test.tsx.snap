// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChipComponent snapshot test 1`] = `
<DocumentFragment>
  <div
    class="MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault undefined border-primary-light bg-primary-light  bg-opacity-25 border-opacity-25 text-primary-dark flex-row-reverse  css-1epr0eg-MuiChip-root"
  >
    <span
      class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
    >
      Test Chip
    </span>
  </div>
</DocumentFragment>
`;
